ckanext/googleanalytics/commands.py

```diff
@@ -238,83 +263,142 @@ def get_dates_between_update(self, start_date, latest_date=None):
+            if not resource:
+                self.log.warning("Couldn't find resource %s" % resource_id)
+                continue
+            for date, value in date_collection.iteritems():
+                ResourceStats.update_visits(resource.id, date, value["downloads"])
+
+    def save_type_package_downloads(self, data):
+        for package_id_or_name, date_collection in data.items():
+            package = model.Package.get(package_id_or_name)
+
+            if not package:
+                self.log.warning("Couldn't find package %s" % package_id_or_name)
+                continue
+            
+            for date, value in date_collection.iteritems():
+                PackageStats.update_downloads(package_id=package.id, visit_date=date, downloads=value["downloads"])
+
+    def save_type_visitorlocation(self, data):
+        for location, visits_collection in data.items():
             visits = visits_collection.get('visits', {})
             for visit_date, count in visits.iteritems():
                 AudienceLocationDate.update_visits(location, visit_date, count)
                 self.log.info("Updated %s on %s with %s visits" % (location, visit_date, count))
 
-    def resolver_type_package(self, querytype, results, data):
+    def resolver_type_package(self, results, data):
         '''
         formats results and returns a dictionary like:
         {
-            'package': { 'cool-dataset-name': { 'visits': { 2019-02-24: 500, ... } }, ... },
+            'package_name_or_id': { 2019-02-24: { 'visits': 500,  'entrances': 400, 'downloads': 300 }},
         }
         '''
         if 'rows' in results:
             for result in results.get('rows'):
-                # this is still specific for packages query
-                package = result[0]
-                # removes /data/ from the url
-                if package.startswith('/data/'):
-                    package = package[len('/data'):]
+                path = result[0]
+                visit_date = datetime.datetime.strptime(result[1], "%Y%m%d").date()
+                
+                splitPath = path.split('/')
+                path_with_vars = splitPath[splitPath.index('dataset') + 1]
+                package_id_or_name = path_with_vars.split('?')[0].split('&')[0]
+
+                visit_count = result[2]
+                entrance_count = result[3]
+
+                # add package_id_or_name if not already there
+                if package_id_or_name not in data:
+                    data.setdefault(package_id_or_name, {})
+                
+                if visit_date not in data[package_id_or_name]:
+                    data[package_id_or_name].setdefault(visit_date, {"visits": 0, "entrances": 0})
+                
+                # Adds visits in different languages together
+                # TODO: check this. does this also add other that are not supposed? Same in downloads
```

Comment:
Do you intend to check this later?

Reply:
Ah yes, forgot to remove that. I made the queries more simpler and so that they only do one thing. This works correctly currently. I'll remove the TODO.